src/Controls/src/Core/Button/Button.iOS.cs

```diff
@@ -14,12 +14,15 @@ namespace Microsoft.Maui.Controls
 {
 	public partial class Button : ICrossPlatformLayout
 	{
+		// _originalImageSize and _originalImageSource are used to ensure we don't resize the image
+		// larger than the original image size and to ensure if a new image is loaded, we use that image's size for resizing.
 		CGSize _originalImageSize = CGSize.Empty;
+		string _originalImageSource = string.Empty;
 
 		// _isFirstMeasure is a flag to make sure we manually recalculate the titleRect when there are dynamic changes to the button.
 		// There are times the platformButton.TitleLabel is updated on dynamic changes and reacts to the change by truncating the label when we actually
 		// have space in our constraints. We provide the space to be used in our first measure of the titleRect and then use the newly laid out titleRect in later iterations.
-		bool _isFirstMeasure = true;
+		internal bool _isFirstMeasure = true;
```

Comment:
Instead of making this internal can we add a method that gets called? 

`ResetToFirstMeasureOfNewContent()`

Or something of that nature?

Reply:
Ah good idea!